text/0046-metadata-for-offline-signers.md

```diff
@@ -53,30 +57,38 @@ Metadata should be stripped from parts that are not necessary to parse a signabl
 
 ## Stakeholders
 
+- Runtime implementors
+- UI/wallet implementors
+- Offline wallet implementors
+
 All chain teams are stakeholders, as implementing this feature would require timely effort on their side and would impact compatibility with older tools.
 
 This feature is essential for **all** offline signer tools; many regular signing tools might make use of it. In general, this RFC greatly improves security of any network implementing it, as many governing keys are used with offline signers.
 
 Implementing this RFC would remove requirement to maintain metadata portals manually, as task of metadata verification would be effectively moved to consensus mechanism of the chain.
 
+The idea for this RFC was brought up by runtime implementors and was extensively discussed with offline wallet implementors. It was designed in such a way that it can work easily with the existing offline wallet solutions in the Polkadot ecosystem.
+
 ## Explanation
 
-Detailed description of metadata shortening and digest process is provided in [metadata-shortener](https://github.com/Alzymologist/metadata-shortener) crate (see `cargo doc --open` and examples). Below are presented algorithms of the process.
+he FRAME metadata is providing a lot of information about a FRAME based runtime. It contains information about the pallets, the calls per pallet, the storage entries per pallet, information about runtime apis and type information about most of the types that are used in the runtime. For decoding transactions on an offline wallet we mainly require these type information. Most of the other information in the FRAME metadata are actually not required for the decoding and thus we can remove them. Thus, we have come up with a custom representation of the metadata and how this custom metadata is chunked, ensuring that we only need to send the chunks requiring for decoding a certain transaction to the offline wallet.
 
-### Definitions
+A reference implementation of this process is provided in [metadata-shortener](https://docs.rs/metadata-shortener/latest/metadata_shortener/) crate.
 
+### Definitions
 
 #### Metadata descriptor
 
 Values for:
 
-1. `u8` metadata shortening protocol version, 
-2. `ExtrinsicMetadata`,
-3. `spec_version` `String`,
-4. `spec_name` `String`,
-5. `u16` ss58 prefix,
-6. `u8` decimals value or `0u8` if no units are defined,
-7. `tokenSymbol` `String` defined on chain to identify the name of currency (available for example through `system.properties()` RPC call) or empty string if no base units are defined,
+1. `u8` metadata shortening protocol version (as encoded enum variant), 
+2. Id of registry type encoding all extrinsic calls (u32),
```

Comment:
I'd point to https://docs.rs/frame-metadata/latest/frame_metadata/v15/struct.ExtrinsicMetadata.html#structfield.call_ty

Reply:
```suggestion
2. The `Id` of type of the outermost Call enum (u32) as described in https://docs.rs/frame-metadata/latest/frame_metadata/v15/struct.ExtrinsicMetadata.html#structfield.call_ty
```